### PR TITLE
add 1d beam

### PR DESCRIPTION
**Pull request by rukawaaaaaaa**
add test 3d slender beam

### CODE REVIEW

File: src/shared/particle_dynamics/slender_structure_dynamics.h [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+		typedef DataDelegateSimple<BarParticles> BarDataSimple;
+		typedef DataDelegateInner<BarParticles> BarDataInner;
+
+		/**
+		 * @class BarDynamicsInitialCondition
+		 * @brief  set initial condition for bar particles
+		 * This is a abstract class to be override for case specific initial conditions.
+		 */
+                class BarDynamicsInitialCondition : public LocalDynamics, public BarDataSimple
+                {
```
Comment (DongWuTUM) [inline src/shared/particle_dynamics/slender_structure_dynamics.h:54]: Should the class name be BarDynamicsInitialCondition or SlenderDynamicsInitialCondition?

**Review by DongWuTUM — COMMENTED** [review]
And all the CI tests should be passed.

File: src/shared/particle_dynamics/slender_structure_math.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+		Vec2d getRotationFromPseudoNormalForSmallDeformation(const Vec2d &dpseudo_n_d2t, const Vec2d &rotation, const Vec2d &angular_vel, Real dt)
+		{
+			return Vec2d(dpseudo_n_d2t[0], 0);
+		}
+		//=================================================================================================//
+        Vec3d getRotationFromPseudoNormalForSmallDeformation(const Vec3d &dpseudo_b_n_d2t, const Vec3d &dpseudo_n_d2t, const Vec3d &rotation, const Vec3d &angular_vel, Real dt)
+		{
+            return Vec3d(0.0, dpseudo_n_d2t[0], 0.0);
+		}
+		//=================================================================================================//
```
Comment (DongWuTUM) [inline src/shared/particle_dynamics/slender_structure_math.cpp:119]: The slender structure is only built up in 3D formulation, right? If it is, please clean all 2D code.
And why is only one angle used to update pseudo-normal?

File: src/shared/particle_dynamics/slender_structure_dynamics.h [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+				Vecd acceleration = Vecd::Zero();
+				Vecd pseudo_normal_acceleration = global_shear_stress_i;
+                Vecd pseudo_b_normal_acceleration = global_b_shear_stress_i;
+
+				const Neighborhood &inner_neighborhood = inner_configuration_[index_i];
+				for (size_t n = 0; n != inner_neighborhood.current_size_; ++n)
+				{
+					size_t index_j = inner_neighborhood.j_[n];
+
+				if (hourglass_control_)
```
Comment (DongWuTUM) [inline src/shared/particle_dynamics/slender_structure_dynamics.h:223]: Seems like the hourglass control algorithm and getCorrectedAlmansiStrin should be modified accordingly.

File: src/shared/particle_dynamics/slender_structure_dynamics.h (hunk 1 above)
Comment (rukawaaaaaaa) [reply src/shared/particle_dynamics/slender_structure_dynamics.h:54]: SlenderStructureDynamicsInitialCondition sounds better and I will change it.

**Comment by Xiangyu-Hu** [comment]
@rukawaaaaaaa seems that the CI not passed.

**Pull request closed without merging**